src/qibolab/instruments/zhinst.py

```diff
@@ -1,15 +1,11 @@
 """Instrument for using the Zurich Instruments (Zhinst) devices."""
 
-import copy
-import os
 from collections import defaultdict
 from dataclasses import dataclass, replace
-from typing import Dict, List, Tuple, Union
+from typing import Dict, Iterable, List, Tuple
```

Comment:
@hay-k, I'm not sure if you're aware or not, but we're not support any longer py3.8, so you could even drop `Dict`/`List`/`Tuple` while you encounter them, and replace with `dict`/`list`/`tuple` builtins.

Also `Iterable` could be replaced, since the new way is to use [`collections.abc.Iterable`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Iterable) (first parametrized type usage of actual classes was unsupported, but after it became supported they started soft-deprecating `typing` hints as duplicated).

Reply:
yes, that's a good point. I am going to polish these type of things as a final step for this PR.